usr/share/rear/restore/BORG/default/400_restore_backup.sh

```diff
@@ -1,30 +1,53 @@
 # This file is part of Relax-and-Recover, licensed under the GNU General
 # Public License. Refer to the included COPYING for full text of license.
 #
-# 400_restore_backup.sk
+# 400_restore_backup.sh
 
 # Borg restores to cwd.
 # Switch current working directory or die.
-pushd $TARGET_FS_ROOT >/dev/null
-StopIfError "Could not change directory to $TARGET_FS_ROOT"
-
-# Start actual restore.
-# Scope of LC_ALL is only within run of `borg extract'.
-# This avoids Borg problems with restoring UTF-8 encoded files names in archive
-# and should not interfere with remaining stages of rear recover.
-# This is still not the ideal solution, but best I can think of so far :-/.
-LogPrint "Recovering from Borg archive $BORGBACKUP_ARCHIVE"
+pushd "$TARGET_FS_ROOT" >/dev/null \
+    || Error "Could not change directory to $TARGET_FS_ROOT!"
 
 # User might specify some additional output options in Borg.
 # Output shown by Borg is not controlled by `rear --verbose' nor `rear --debug'
-local borg_additional_options=''
+# only, if BORGBACKUP_SHOW_PROGRESS is true.
+
+# shellcheck disable=SC2168
+local borg_additional_options=()
 
-is_true $BORGBACKUP_SHOW_PROGRESS && borg_additional_options+='--progress '
+BORGBACKUP_EXTRACT_SHOW_PROGRESS=${BORGBACKUP_EXTRACT_SHOW_PROGRESS:-$BORGBACKUP_SHOW_PROGRESS}
+BORGBACKUP_EXTRACT_SHOW_LIST=${BORGBACKUP_EXTRACT_SHOW_LIST:-$BORGBACKUP_SHOW_LIST}
+BORGBACKUP_EXTRACT_SHOW_RC=${BORGBACKUP_EXTRACT_SHOW_RC:-$BORGBACKUP_SHOW_RC}
 
-LC_ALL=rear.UTF-8 \
-borg extract --sparse $borg_additional_options $BORGBACKUP_OPT_REMOTE_PATH \
-${borg_dst_dev}${BORGBACKUP_REPO}::$BORGBACKUP_ARCHIVE 0<&6 1>&7 2>&8
+is_true "$BORGBACKUP_SHOW_PROGRESS" && borg_additional_options+=( --progress )
+is_true "$BORGBACKUP_SHOW_LIST" && borg_additional_options+=( --list )
+is_true "$BORGBACKUP_SHOW_RC" && borg_additional_options+=( --show-rc )
+
+# https://github.com/rear/rear/pull/2382#issuecomment-621707505
+# Depending on BORGBACKUP_SHOW_PROGRESS and VERBOSE variables
+# 3 cases are there for `borg_extract` to log to rear log file or not.
+#
+# 1. BORGBACKUP_SHOW_PROGRESS true:
+#    No logging to rear log file because of control characters.
+#
+# 2. VERBOSE true:
+#    stderr (2) is copied to real stderr (8):
+#    2 is going to rear logfile
```

Comment:
The comment does no longer match what the code does because
now both stdout and stderr are copied to the ReaR log file
and redirected to the user's terminal.

Reply:
See a5948d5578bd5ab9155a4650259b8eed34e79c74.